compiler-core/src/compiler_cache.rs

```diff
@@ -0,0 +1,122 @@
+use crate::type_::{Type, ValueConstructorVariant};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::sync::Arc;
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct Module {
+    pub name: Vec<String>,
+    pub types: HashMap<String, TypeConstructor>,
+    pub values: HashMap<String, ValueConstructor>,
+    pub accessors: HashMap<String, AccessorsHashMap>,
+    pub package: String,
+    pub types_constructors: HashMap<String, Vec<String>>,
+}
+
+// #[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+// pub enum Type {
+//     Var(u64),
+//     Tuple(Vec<Type>),
+//     App {
+//         name: String,
+//         module: Vec<String>,
+//         parameters: Vec<Type>
+//     },
+//     Fn {
+//         arguments: Vec<Type>,
+//         _return: Type
+//     },
+// }
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct TypeConstructor {
+    pub _type: Arc<Type>,
+    // TODO: convert this to an int as we only need to reconstruct type vars,
+    // not other types
+    // TODO: test
```

Comment:
Copied from the cap'n schema, what does it mean by convert to int

Reply:
Oh yes! It means that these do not need to be types as they are parameter definitions here. All we need to know is which ones are the same, because they are always Type::Var(Generic)

Reply:
So we just take an integer? which represents each type?

Reply:
compiler cache file gone